### PR TITLE
feat: add polygon mapper and parcelization doc

### PR DESCRIPTION
**Pull request by Dinasoa**

### CODE REVIEW

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 1]
```diff
@@ -35,4 +43,63 @@ public Parcel toDomain(
...
+    List<Coordinate> polygonCoords = new ArrayList<>();
+
+    if (multiPolygonCoordinates != null && !multiPolygonCoordinates.isEmpty()) {
+      //                [[[x,y],[x,y],[x,y]]]: Polygon
+      List<List<List<BigDecimal>>> polygon = multiPolygonCoordinates.get(0);
+
+      for (List<List<BigDecimal>> ring : polygon) {
+        //                [[x,y],[x,y],[x,y]]: Ring
+        Coordinate[] ringCoords = new Coordinate[ring.size()];
+        System.out.println("Taille d'un ring: " + ring.size());
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: remove "System.out.println", use `@Slf4j` if necessary

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 2]
```diff
@@ -35,4 +43,63 @@ public Parcel toDomain(
...
   }
+
+  public static Polygon toDomain(Feature feature) {
+    var multiPolygonCoordinates = feature.getGeometry().getCoordinates();
+
+    GeometryFactory geometryFactory = new GeometryFactory();
+
+    List<Coordinate> polygonCoords = new ArrayList<>();
+
+    if (multiPolygonCoordinates != null && !multiPolygonCoordinates.isEmpty()) {
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: throw exception immediatly on empty or null

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 3]
```diff
@@ -35,4 +43,63 @@ public Parcel toDomain(
...
+      //                [[[x,y],[x,y],[x,y]]]: Polygon
+      List<List<List<BigDecimal>>> polygon = multiPolygonCoordinates.get(0);
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: suspicious. Use getFirst() if really necessary

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 4]
```diff
@@ -35,4 +43,63 @@ public Parcel toDomain(
...
+
+      for (List<List<BigDecimal>> ring : polygon) {
+        //                [[x,y],[x,y],[x,y]]: Ring
+        Coordinate[] ringCoords = new Coordinate[ring.size()];
+        System.out.println("Taille d'un ring: " + ring.size());
+        for (int i = 0; i < ring.size(); i++) {
+          BigDecimal x = ring.get(i).get(0);
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: getFirst()

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 5]
```diff
@@ -35,4 +43,63 @@ public Parcel toDomain(
...
+          BigDecimal y = ring.get(i).get(1);
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: getLast() as ring size is equal to 2

**Review by nvagno — CHANGES_REQUESTED** [review]

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 6]
```diff
@@ -35,4 +43,54 @@ public Parcel toDomain(
...
+    var multiPolygonCoordinates = feature.getGeometry().getCoordinates();
+    GeometryFactory geometryFactory = new GeometryFactory();
+    List<Coordinate> polygonCoords = new ArrayList<>();
+
+    if (multiPolygonCoordinates == null || multiPolygonCoordinates.isEmpty()) {
+      throw new IllegalArgumentException("Multipolygon coordinates should not be null");
+    }
+
+    List<List<List<BigDecimal>>> polygon = multiPolygonCoordinates.getFirst();
+    for (List<List<BigDecimal>> ring : polygon) {
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: Use for each instead

**Review by nvagno — CHANGES_REQUESTED** [review]

File: doc/api.yml [hunk 7]
```diff
@@ -464,7 +464,29 @@ paths:
                 $ref: "#/components/schemas/ZoneDetectionJob"
         '500':
           $ref: '#/components/responses/500'
-
+  '/parcelize':
```
Comment (bp-lou) [inline doc/api.yml]: No! Never use a verb as endpoint! Create nouns and objects, eg: parcelization

File: src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java [hunk 8]
```diff
@@ -35,4 +44,57 @@ public Parcel toDomain(
   public static Feature from(TilingTask domainTask) {
     return domainTask.getParcelContent().getFeature();
   }
+
+  public Polygon toDomain(Feature feature) {
+    List<List<List<List<BigDecimal>>>> multiPolygonCoordinates =
+        feature.getGeometry().getCoordinates();
+
```
Comment (bp-lou) [inline src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/FeatureMapper.java]: superfluous empty line

**Review by bp-lou — APPROVED** [review]

**Pull request merged**